cli/src/cmd/forge/generate/mod.rs

```diff
@@ -0,0 +1,69 @@
+//! generate command
+
+use clap::{Parser, Subcommand};
+use std::{fs, path::Path};
+
+/// CLI arguments for `forge generate`.
+#[derive(Debug, Parser)]
+pub struct GenerateArgs {
+    #[clap(subcommand)]
+    pub sub: GenerateSubcommands,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum GenerateSubcommands {
+    /// Scaffolds test file for given contract.
+    Test(GenerateTestArgs),
+}
+
+#[derive(Debug, Parser)]
+pub struct GenerateTestArgs {
+    /// Contract name for test generation.
+    #[clap(long, short, value_name = "CONTRACT_NAME")]
+    pub contract_name: String,
+}
+
+impl GenerateTestArgs {
+    pub fn run(self) -> eyre::Result<()> {
+        let contract_name = format_identifier(&self.contract_name, true);
+        let instance_name = format_identifier(&self.contract_name, false);
+
+        // Create the test file content.
+        let test_content = include_str!("../../../../assets/generated/TestTemplate.t.sol");
+        let test_content = test_content
+            .replace("{contract_name}", &contract_name)
+            .replace("{instance_name}", &instance_name);
+
+        // Create the test directory if it doesn't exist.
+        fs::create_dir_all("test").unwrap();
```

Comment:
Could we avoid `unwrap` here? We can just use `?` and maybe `map_err` as well to make it a tad more descriptive.

Reply:
Absolutely! How does the following code look?

```rust
    // Create the test directory if it doesn't exist
    fs::create_dir_all("test").map_err(|e| {
        format!(
            "Failed to create 'test' directory: {}. Please check if you have write access.",
            e
        )
    });
```

Reply:
Absolutely! I looked at some other files and figured there is wrapper around fs `foundry_common::fs` so I have swapped the implementation.